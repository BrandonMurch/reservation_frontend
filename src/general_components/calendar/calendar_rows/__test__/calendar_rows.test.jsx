// Dependencies
import React from 'react';
import { unmountComponentAtNode } from 'react-dom';
import { render } from '@testing-library/react';
import { create } from 'react-test-renderer';
import moment from 'moment';

// Components
import Row from '../index';

describe('<CalendarRow />', () => {
  let component;
  let container = null;
  const dateObject = moment();
  const mockRenderFunction = jest.fn(({ setMessage }) => { setMessage('hello'); });

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
    component = render(
      <table>
        <Row dateObject={dateObject} onDateRender={mockRenderFunction} />
      </table>,
    );
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it('should match snapshot', () => {
    const tree = create(<Row dateObject={dateObject} onDateRender={mockRenderFunction} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have rendered cells', () => {
    const cells = component.getAllByRole('gridcell');
    expect(cells.length).toBeGreaterThan(dateObject.daysInMonth());
  });

  it('should call onDateRender prop', () => {
    expect(mockRenderFunction).toHaveBeenCalled();
  });
});
