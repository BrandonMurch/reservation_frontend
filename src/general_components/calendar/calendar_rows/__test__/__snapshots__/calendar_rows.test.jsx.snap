// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalendarRow /> should match snapshot 1`] = `
<tbody
  className="calendarBody"
>
  <tr />
  <tr>
    <td
      className="container"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        
      </p>
      <p
        className="messageText"
      />
    </td>
    <td
      className="container"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        
      </p>
      <p
        className="messageText"
      />
    </td>
    <td
      className="container"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        
      </p>
      <p
        className="messageText"
      />
    </td>
    <td
      className="container"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        
      </p>
      <p
        className="messageText"
      />
    </td>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        1
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        2
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        3
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
  </tr>
  <tr>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        4
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        5
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        6
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        7
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        8
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        9
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        10
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
  </tr>
  <tr>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        11
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        12
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        13
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        14
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        15
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        16
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        17
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
  </tr>
  <tr>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        18
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        19
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        20
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        21
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        22
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        23
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        24
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
  </tr>
  <tr>
    <td
      className="todayContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        25
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        26
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        27
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        28
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        29
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        30
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
    <td
      className="activeContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="gridcell"
    >
      <p
        className="dateText"
      >
        31
      </p>
      <p
        className="messageText"
      >
        hello
      </p>
    </td>
  </tr>
</tbody>
`;
